src/ripple/app/tx/impl/NFTokenAcceptOffer.cpp

```diff
@@ -0,0 +1,366 @@
+        // cut, if any).
+        if (auto const brokerFee = ctx.tx[~sfBrokerFee])
+        {
+            if (brokerFee->issue() != (*bo)[sfAmount].issue())
+                return tecBUY_SELL_MISMATCH;
+
+            if (brokerFee >= (*bo)[sfAmount])
+                return tecINSUFFICIENT_PAYMENT;
+
+            if ((*so)[sfAmount] > (*bo)[sfAmount] - *brokerFee)
+                return tecINSUFFICIENT_PAYMENT;
+        }
+    }
+
+    if (buy)
+    {
+        auto const bo = ctx.view.read(keylet::nftoffer(*buy));
+
+        if (((*bo)[sfFlags] & lsfSellToken) == lsfSellToken)
+            return tecOFFER_TYPE_MISMATCH;
+
+        // An account can't accept an offer it placed:
+        if ((*bo)[sfOwner] == ctx.tx[sfAccount])
+            return tecCANT_ACCEPT_OWN_OFFER;
+
+        // If not in bridged mode, the account must own the token:
+        if (!sell &&
+            !nft::findToken(ctx.view, ctx.tx[sfAccount], (*bo)[sfTokenID]))
+            return tecNO_PERMISSION;
+
+        // The account offering to buy must have funds:
+        auto const needed = bo->at(sfAmount);
+
+        if (accountHolds(
+                ctx.view,
+                (*bo)[sfOwner],
+                needed.getCurrency(),
+                needed.getIssuer(),
+                fhZERO_IF_FROZEN,
+                ctx.j) < needed)
+            return tecINSUFFICIENT_FUNDS;
+    }
+
+    if (sell)
+    {
+        auto const so = ctx.view.read(keylet::nftoffer(*sell));
+
+        if (((*so)[sfFlags] & lsfSellToken) != lsfSellToken)
+            return tecOFFER_TYPE_MISMATCH;
+
+        // An account can't accept an offer it placed:
+        if ((*so)[sfOwner] == ctx.tx[sfAccount])
+            return tecCANT_ACCEPT_OWN_OFFER;
+
+        // The seller must own the token.
+        if (!nft::findToken(ctx.view, (*so)[sfOwner], (*so)[sfTokenID]))
+            return tecNO_PERMISSION;
+
+        // If not in bridged mode...
+        if (!buy)
+        {
+            // If the offer has a Destination field, the acceptor must be the
+            // Destination.
+            if (auto const dest = so->at(~sfDestination);
+                dest.has_value() && *dest != ctx.tx[sfAccount])
+                return tecNO_PERMISSION;
+        }
+
+        // The account offering to buy must have funds:
+        auto const needed = so->at(sfAmount);
+
+        if (accountHolds(
+                ctx.view,
+                ctx.tx[sfAccount],
+                needed.getCurrency(),
+                needed.getIssuer(),
+                fhZERO_IF_FROZEN,
+                ctx.j) < needed)
+            return tecINSUFFICIENT_FUNDS;
+    }
+
+    return tesSUCCESS;
+}
+
+TER
+NFTokenAcceptOffer::pay(
+    AccountID const& from,
+    AccountID const& to,
+    STAmount amount)
+{
+    // This should never happen, but it's easy and quick to check.
+    if (amount < beast::zero)
+        return tecINTERNAL;
+
+    return accountSend(view(), from, to, amount, j_);
```

Comment:
I _think_ this is okay as written, but maybe you'll convince me otherwise.  Let me walk through the steps.

1. The total paid for the NFT always equals or exceeds all fees associated with the transfer -- both the transfer fee and the broker fee.
2. If there is a broker, the broker is paid their fee first.  That is subtracted from the total paid.
3. If the seller specified a selling price then the remainder, after the broker is paid, must equal or exceed the specified selling price.
4. The issuer's portion is determined as a percentage of this remainder.  We now pay that portion to the issuer.  The amount paid to the issuer is then subtracted from the total paid.
5. At this point, the last bit of the remainder may be _below_ the amount requested by the seller.  We consider that to be okay because the seller is paying the transfer fee to the issuer.
6. These last remaining funds are then transferred to the seller.

So the effect is that the seller is paying the transfer fee, even though the funds did not actually transfer from buyer to seller to issuer.  The final effect is that the seller is paying the transfer fee to the issuer.

I'm open to further discussion if this doesn't seem right to you.